prow/plugins/trigger/push.go

```diff
@@ -73,20 +95,29 @@ func handlePE(c Client, pe github.PushEvent) error {
 	}
 
 	postsubmits := getPostsubmits(c.Logger, c.GitClient, c.Config, org+"/"+repo, shaGetter)
+	prs, err := getPullRequests(c.GitHubClient, pe)
+	if err != nil {
+		return err
+	}
 
 	for _, j := range postsubmits {
 		if shouldRun, err := j.ShouldRun(pe.Branch(), listPushEventChanges(pe)); err != nil {
 			return err
 		} else if !shouldRun {
 			continue
 		}
-		refs := createRefs(pe)
-		labels := make(map[string]string)
-		for k, v := range j.Labels {
-			labels[k] = v
+		var pj prowapi.ProwJob
+		if len(prs) > 0 {
+			pj = pjutil.NewPostsubmit(prs[0], prs[0].Base.SHA, j, pe.GUID)
```

Comment:
I wouldn't trust GitHub's api to provide consistent ordering of PR's associated with a commit

I would at least recommend filtering down to whichever PR has most recently been merged

Reply:
Thank you for the review. Fixed.

Reply:
[Postsubmits default to always-run][1], do you know if this means they're always going to report to PRs as well?  That would probably be an unwelcome default for something like kubernetes/kubernetes

[1]: https://github.com/kubernetes/test-infra/blob/0214d73dd35b87730d5fa05d713c81c9404b51fa/prow/config/jobs.go#L330-L331